### DISCUSSION TITLE
Logging acked messages before they are removed

### DISCUSSION THREAD

**Discussion by zaeemarshad** [rabbitmq/rabbitmq-server#7449] [Questions] [answered]
Is it possible for the queue process to emit a log for each message that is ACK'd or dropped from the queue? I assume there will be performance implications but perhaps as a DEBUG level this may work.

**Comment by michaelklishin** [comment, marked as answer]
RabbitMQ never logs per-message operations unless they result in a channel or connection exception.

There is a tracing mechanism: https://github.com/MarcialRosales/rabbitmq-tracing-guide but it is a specialized debug tool.

A traffic capture would work even better than tracing.

**Reply by zaeemarshad** [reply]
Thanks heaps.
